### ISSUE TITLE
Firebase Webpage Not Receiving Data

### ISSUE THREAD

**Issue by aaronburnie** [Leverege/microchip-avr-iot#12]
Hello,

I am hoping someone could point me in the right direction to get my Google Cloud webpage to display data from my AVR-IoT device.
I have successfully setup a project within the IoTCore which receives data packets from my IoT device.
│ {"Light":3,"Temp":"25.75"} │ 

To view data within the cloud I setup a pub/sub subscription to topic avr-iot which successfully shows the data received from the IoT device.

However, when I navigate to the firebase website https://{project}.firebaseapp.com/device/{device ID} it cannot connect to the device and retrieve data.
![Webapp](https://user-images.githubusercontent.com/45171644/60754421-3a08f880-a013-11e9-97dd-048d66acc6b2.png)

Any ideas on what is going wrong here.
Thanks in advance.

**Comment by agronc** [comment]
Can you see the data within your firebase database?

**Comment by aaronburnie** [comment]
Didn't know a firebase database was required....
I was just following the microchip-avr-iot instructions which did not mention anything about a database. If a database is required do you know what the requirements are? For example, is a specific collection ID required?

**Comment by agronc** [comment]
The script is supposed to create that database for you automatically (as far as I know) though that is not occurring for me any longer. You can check in your firebase account, under your project, then databases. If you dont see the database there, then you have the same issue as me, if you do see it, then might be something else.

**Comment by aaronburnie** [comment]
Thanks for the reply,
A database definitely hasn't been created within Firebase. However, I believe the problem you are seeing is slightly different than mine. 
I can see data within the IoTCore using pub/sub commands, whereas I believe you cannot.

I shall review the script and see if I can create a database myself to get it working.

**Comment by agronc** [comment]
can I ask what commands are you using to check this data? I will check it on my end as well.

**Comment by aaronburnie** [comment]
I believe the best way to do this would be to follow the interactive tutorial "Interactive Tutorial in GCP Console" at the link https://cloud.google.com/pubsub/docs/tutorials, this is what I did. As you go through the tutorial you can adjust the commands to look for data within the IoT project.

You should only need one command after running the AVR-IoT script:
gcloud pubsub subscriptions create iot-sub --topic avr-iot `--ack-deadline=60

This command creates a subscription "iot-sub" attached to topic "avr-iot". A subscription allows you to retrieve messages from a topic. You can then use the following command to pull data:
gcloud pubsub subscriptions pull --auto-ack avr-iotsub

However, within Google Cloud Platform -> IoT Core -> Monitoring the graphs should indicate that data is already being received. pubsub is just a way to view the data.

**Comment by aaronburnie** [comment]
I had a win following the recommendations posted in your raised issue (@agronc). 
> Node.js 6 has deprecated you need to update your code.

All packages were updated using npm update. This was done in both ui and function folders.
I added the following lines to ./functions/package.json after receiving _Error: There was an error reading ./functions/package.json:_
 `"engines": {`
 `"node": "8"`
  `},`
After working through the updates I got: 
`Deploy complete! with no errors`

Firebase was properly configured and a Realtime Database created. I can now navigate to my device webpage and see data.
